src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -178,6 +178,48 @@ internal void UpdateHtmlBuffer(UpdateBufferRequest request)
                 request.HostDocumentVersion.Value);
         }
 
+        public override async Task<RazorDocumentRangeFormattingResponse> RazorDocumentFormattingAsync(DocumentFormattingParams request, CancellationToken cancellationToken)
+        {
+            var response = new RazorDocumentRangeFormattingResponse() { Edits = Array.Empty<TextEdit>() };
+
+            await _joinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
So interesting, I wonder why we jumped to main thread in the range formatting variant. Does anything below require the main thread?

Reply:
Looks like it was instead of using a semaphore to sync access to the buffer?
https://github.com/dotnet/aspnetcore-tooling/pull/1961/files#diff-4ae161f8c61872340d5acd2d4f77d823d9cc550ecc086581222b5d953349bc3aL75-R70

Reply:
Hmm, one of these days i'll have to dig through these bits to understand why the semaphore slim here was even required. Problem for another day I guess 😄